From 23712cea8e2a623fd952eb781df0011c501703d0 Mon Sep 17 00:00:00 2001
From: Thorsten Glaser <tg@mirbsd.org>
Date: Thu, 25 Jul 2013 22:07:33 +0200
Subject: [PATCH] Make default mkshrc file suitable for OpenWrt environment:

* Part of the FreeWRT patches:
  - no hostname(1)
  - ls(1) has no -o option
* OpenWrt and FreeWRT-1.0 fix:
  - since this is not ~/.mkshrc make sure subshells find it

From: Alif M. A. <alive4ever@live.com>
Date: Thu, 20 Jul 2017 14:52:39 +0000
Subject: [PATCH] Refresh 100-dot_mkshrc for mksh-R55

Additional changes of the patch as of mksh-R55
* Use content of `/proc/sys/kernel/hostname` to set `$HOSTNAME`, since
  `hostname` applet isn't compiled into `busybox`.
* Use `/bin/vi` as fallback `$EDITOR`.
* Use `/etc/mkshrc` as default startup file, so there is no need to
  manually source it during interactive session.

From: Alif M. A. <alive4ever@live.com>
Date: Thu, 11 Jan 2018 02:13:46 +0000
Subject: [PATCH] Refresh 100-dot_mkshrc for mksh-R55

Refreshed 100-dot_mkshrc for mksh-r56b

---
Reviewed-by: Thorsten Glaser <tg at mirbsd.org>
Signed-off-by: Alif M. A. <alive4ever at live.com>

--- a/dot.mkshrc
+++ b/dot.mkshrc
@@ -64,9 +64,9 @@ for EDITOR in "${EDITOR:-}" jupp jstar m
 	EDITOR=
 done
 
-\\builtin alias ls=ls l='ls -F' la='l -a' ll='l -l' lo='l -alo'
-\: "${EDITOR:=/bin/ed}${TERM:=vt100}${USER:=$(\\builtin ulimit -c 0; id -un \
-    2>/dev/null)}${HOSTNAME:=$(\\builtin ulimit -c 0; hostname 2>/dev/null)}"
+\\builtin alias ls=ls l='ls -F' la='l -a' ll='l -l' lo='l -al'
+\: "${EDITOR:=/bin/vi}${TERM:=vt100}${USER:=$(\\builtin ulimit -c 0; id -un \
+    2>/dev/null)}${HOSTNAME:=$(</proc/sys/kernel/hostname)}"
 [[ $HOSTNAME = ?(?(ip6-)localhost?(6)) ]] && HOSTNAME=nil; \\builtin unalias ls
 \\builtin export EDITOR HOSTNAME TERM USER="${USER:-?}"
 
